OOP_Homework_2/BankCount.cs

```diff
@@ -102,6 +102,60 @@ public void PrintCount()
             Console.WriteLine($" Ваш тип счета: {typeOfCount}\n Ваш номер счета: {accountNumber}\n Ваш баланс: {balance}\n ");
 
         }
+
+        public override bool Equals(object obj)
+        {
+            BankCount acc;
+            try
+            {
+                acc = (BankCount)obj;
+            }
+            catch
+            {
+                return false;
+            }
+            return this.accountNumber == acc.accountNumber;
+        }
+        /// <summary>
+        /// Переопределеный оператор для сравнения двух аккаунтов
+        /// </summary>
+        /// <param name="mycount1">акк1</param>
+        /// <param name="mycount2">акк2</param>
+        /// <returns></returns>
+        public static bool operator == (BankCount mycount1, BankCount mycount2)
+        {
+            return Equals(mycount1, mycount2);
+        }
+        /// <summary>
+        /// Переопределеный оператор для сравнения двух аккаунтов
+        /// </summary>
+        /// <param name="mycount1">акк1</param>
+        /// <param name="mycount2">акк2</param>
+        /// <returns></returns>
+        public static bool operator != (BankCount mycount1, BankCount mycount2)
+        {
+            return !Equals(mycount1, mycount2);
+        }
+        /// <summary>
+        /// Перегруженный метод, возвращает хеш-код аккаунта
+        /// </summary>
+        /// <returns></returns>
+        public override int GetHashCode()
+        {
+            return count + accountNumber;
```

Comment:
Если на скорую руку, то сойдёт. Но я бы не советовал. Если лень писать полноценный алгоритм вычисления хеша, то можно использовать готовый инструмент для этого.
```suggestion
            return HashCode.Combine(count, accountNumber);
```

Reply:
Да я почитал в документации, буду использовать. Спасибо